### DISCUSSION TITLE
Cannot create empty file with "Update a gist" API

### DISCUSSION THREAD

**Discussion by wh201906** [community/community#58426] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

Hi. I'm trying to create a empty file with "Update a gist" API, but the server returns 422
```
request: b'{"files": {"wh201906#StarTracker": {"content": ""}}}'
response(422): {"message":"Validation Failed","errors":[{"resource":"Gist","code":"missing_field","field":"files"}],"documentation_url":"https://docs.github.com/rest/reference/gists/#update-a-gist"}
```
I cannot even create a file with a simple empty line
```
request: b'{"files": {"wh201906#StarTracker": {"content": "\\n"}}}'
response(422): {"message":"Validation Failed","errors":[{"resource":"Gist","code":"missing_field","field":"files"}],"documentation_url":"https://docs.github.com/rest/reference/gists/#update-a-gist"}
```
However, after I change the content field into `"(no users)\n"`, the server returns 200

**Comment by wh201906** [comment]
Is there anything else I need to provide? I saw this discussion was labeled with `more information needed`

**Comment by byrneh** [comment]
@wh201906 The simple answer is you cannot create an "empty" gist. and the API validation is checking and preventing a user accidentally doing so. and this behaviour feels reasonable to me.  
The GUI also prevents empty gists.  The "422" reposne to your API call is for the validation error.

![image](https://github.com/community/community/assets/39440862/150f5492-a044-491b-bc3e-b133f11b8152)

**Comment by wh201906** [comment]
@byrneh Thanks for your answer!
I think calling the "Update a gist" API should have different behavior from creating a gist from webpage, whose corresponding API is "Create a gist". Plus, the "Update a gist" API allows users to delete a file in a gist.
https://docs.github.com/en/rest/gists/gists?apiVersion=2022-11-28#create-a-gist
https://docs.github.com/en/rest/gists/gists?apiVersion=2022-11-28#update-a-gist

Your answer shows creating a file with a simple empty line might be treated as creating an empty file. That's reasonable. But I still think the behavior I mentioned above is not intuitive.

**Reply by byrneh** [reply]
Creating an empty gist rather than just deleting it seems unnecessary to me when the common use for a gist is to share information and sharing nothing seems counter-intuitive to me.

Your only option currently though  seems to be to leave some minimal content in the file, say a single full stop.
If the ability to be able to update a gist with empty content remains impactful for your needs you could try raising a bug/feature request so that sometime in the future you will be abke to do what you ask. 
Or updating an exiting gist to empty in the GUI.

**Reply by wh201906** [reply]
> Your only option currently though seems to be to leave some minimal content in the file, say a single full stop. 

Yes. That's the workaround I'm using now.

> you could try raising a bug/feature request so that sometime in the future you will be abke to do what you ask.

IDK if this post there is a bug report/feature request. Should I raise it somewhere else?
